pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -342,7 +342,7 @@ func (og *operationGenerator) GenerateAttachVolumeFunc(
 				uncertainNode = derr.CurrentNode
 			}
 			addErr := actualStateOfWorld.MarkVolumeAsUncertain(
```

Comment:
How did you address the race we discussed? CSIDriver object is created (with `skipattach`) during first attach attempt, so uncertain state fails to set.

Reply:
This is a very interesting point. In fact, if we pass the volumeName to the MarkVolumeAsUncertain() function, the markasuncertain operation will not fail. So in that case, the race will not exist because MarkVolumeAsUncertain will always be able to happen no matter when do we deploy the CSIDriver. That's why I am curious about why we are giving an empty volumeName. 

If we have to keep the empty volume name for some reason. What will happen is that the MarkVolumeAsUncertain will fail. And the volumeattachment object will be long existing. But the pod is able to bringup because kubelet will detect its attachability change.